### PR TITLE
Install Chakra UI

### PR DESCRIPTION
**Pull request by carolynzhang18**
## Notion Ticket Link

[Install Chakra UI](https://www.notion.so/uwblueprintexecs/Install-Chakra-UI-86ca298de7e940bc8a148c9001761e22)

## Implementation Description
* The actual installation changes are mostly in the frontend package.json and yarn.lock. The rest of the changes were just small fixes to accommodate the updated packages.

## Steps to Test
1. Delete your `frontend/node_modules` folder. In the frontend directory (`cd frontend`), run `yarn install` to get a fresh copy of the packages. This won't affect what Docker gives you, but it'll make the local package errors go away.
2. After running `docker compose down`, rebuild the containers with `docker compose up --build`.
3. Expected: app compiles with no errors*.

\* The login page will probably show "Script error" (and you'll see something about missing the client_id parameter in the console). Don't worry about it, the "Remove Google Auth for Login" ticket will make the error go away eventually.

## Screenshots (optional)
Here's what the Chakra UI button looks like:
image
If you want to also plop a component somewhere to try it out, check out the docs: https://chakra-ui.com/docs/components

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### Mac
```bash
docker exec -it CWC-backend /bin/bash -c "yarn fix"
docker exec -it CWC-frontend /bin/bash -c "yarn fix"
```

### Windows
```bash
docker exec -it CWC-backend bash -c "yarn fix"
docker exec -it CWC-frontend bash -c "yarn fix"
```

### CODE REVIEW

File: frontend/src/components/auth/Login.tsx [hunk 1]
```diff
@@ -42,17 +42,17 @@ const Login = (): React.ReactElement => {
...
       email,
       password,
       login,
     );
     setAuthenticatedUser(user);
+    if (isVerified === false) {
+      window.alert(
+        "Failed to log in. Please check your email for a link to verify your account.",
+      );
+    }
```
Comment (carolynzhang18) [inline frontend/src/components/auth/Login.tsx:55]: This change is unrelated to the ticket... you might've noticed there's sometimes a little glitch with this alert popping up even when you log in with a verified account - the state of this component's a bit out of sync, so I was trying to move the alert down to fix it... but that didn't work lol we can make a bug ticket for this at some point

File: frontend/.eslintrc.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ module.exports = {
     "react/no-array-index-key": "off",
     "jsx-a11y/click-events-have-key-events": "off",
     "jsx-a11y/no-static-element-interactions": "off",
+    "@typescript-eslint/no-useless-constructor": "off",
```
Comment (owen-sellner) [inline frontend/.eslintrc.js:34]: What does this do?

File: frontend/src/APIClients/SimpleEntityAPIClient.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ const create = async ({
     });
     return data;
   } catch (error) {
-    return error;
+    return error as SimpleEntityResponse;
```
Comment (owen-sellner) [inline frontend/src/APIClients/SimpleEntityAPIClient.ts:44]: I am not a big fan of this but I am not sure what the better alternative is :/

**Review by owen-sellner — COMMENTED** [review]
I realized while testing your code that the functions in `frontend/src/APIClients` are never called in our application 😭

File: frontend/src/components/auth/Login.tsx (hunk 1 above)
Comment (owen-sellner) [reply frontend/src/components/auth/Login.tsx:56]: The issue can be resolved by adding an await statement before isVerified. Also, please check if there are any other places with the same issue. 🙏 
```
if ((await isVerified) === false) {
      // eslint-disable-next-line no-alert
      window.alert(
        "Failed to log in. Please check your email for a link to verify your account.",
      );
}

File: frontend/src/APIClients/SimpleEntityAPIClient.ts (hunk 3 above)
Comment (owen-sellner) [reply frontend/src/APIClients/SimpleEntityAPIClient.ts:44]: I spoke with the VP of Engineering, and they think the best thing here would be to throw the error instead of trying to return something. 

```
catch (error: unknown) {
    throw error;
 }

**Review by owen-sellner — CHANGES_REQUESTED** [review]
I left some comments about the error handling and alert issues 🫡

File: frontend/.eslintrc.js (hunk 2 above)
Comment (carolynzhang18) [reply frontend/.eslintrc.js:34]: There's one file that has a constructor eslint deems useless/unnecessary. I don't remember which file exactly, but I'm pretty sure it was in node_modules somewhere... so not even our own code. I'll check again to see if there's any other workaround!

File: frontend/src/APIClients/SimpleEntityAPIClient.ts (hunk 3 above)
Comment (carolynzhang18) [reply frontend/src/APIClients/SimpleEntityAPIClient.ts:44]: Ah okay sounds good! This is definitely something that should get fixed in the starter code... it's weird to see a function returning its error instead of throwing it...

File: frontend/.eslintrc.js (hunk 2 above)
Comment (carolynzhang18) [reply frontend/.eslintrc.js:34]: Woah I just ran it without that line and it compiled! guess I changed some package between now and when I had that error and accidentally fixed it lol 😅
thanks for catching this!!

File: frontend/src/APIClients/SimpleEntityAPIClient.ts (hunk 3 above)
Comment (carolynzhang18) [reply frontend/src/APIClients/SimpleEntityAPIClient.ts:44]: Update: this makes eslint throw an error 🤨

Comment (carolynzhang18) [reply frontend/src/APIClients/SimpleEntityAPIClient.ts:44]: Ok I fixed the linter problems now :D
Linter doesn't love `catch (error) { throw error; }` - it says the try-catch block is unnecessary if we're just going to throw it without performing any other actions...

File: frontend/src/APIClients/EntityAPIClient.ts [hunk 4]
```diff
@@ -36,64 +36,46 @@ const create = async ({
...
   )}`;
-  try {
-    const { data } = await baseAPIClient.post("/entities", formData, {
-      headers: { Authorization: bearerToken },
-    });
-    return data;
-  } catch (error) {
-    return error;
-  }
+  const { data } = await baseAPIClient.post("/entities", formData, {
```
Comment (owen-sellner) [inline frontend/src/APIClients/EntityAPIClient.ts:39]: We still need to catch the error to ensure it does not propagate up the stack when this is called. In `AuthAPIClient.ts` the functions return null if an error is caught (ex. `login`). 

We could do something like this:
```
const create = async ({
  formData,
}: {
  formData: FormData;
}): Promise<EntityResponse | null> => {
  const bearerToken = `Bearer ${getLocalStorageObjProperty(
    AUTHENTICATED_USER_KEY,
    "accessToken",
  )}`;
  try {
    const { data } = await baseAPIClient.post("/entities", formData, {
      headers: { Authorization: bearerToken },
    });
    return data;
  } catch (e: unknown) {
    return null;
  }
};

**Comment by SohailSayed** [comment]
Followed the steps, had no issues compiling on my end

**Review by debs415 — APPROVED** [review]
Great work on this Chakra UI ticket! The changes look good, and following the steps you laid out to test, I could run smoothly and get a few Chakra components added to the login page as well, so it seems to work. Once again, great job on this one! :)

**Review by SohailSayed — APPROVED** [review]

**Review by owen-sellner — APPROVED** [review]
I double-checked that the try-catch statement passes the linter. This is ready to merge, good job!!!
